src/NzbDrone.Core/MediaFiles/EpisodeImport/Aggregation/Aggregators/Augmenters/Quality/AugmentQualityFromReleaseName.cs

```diff
@@ -0,0 +1,58 @@
+using System.Linq;
+using NzbDrone.Core.Download;
+using NzbDrone.Core.History;
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.Qualities;
+
+namespace NzbDrone.Core.MediaFiles.EpisodeImport.Aggregation.Aggregators.Augmenters.Quality
+{
+    public class AugmentQualityFromReleaseName : IAugmentQuality
+    {
+        private readonly IHistoryService _historyService;
+
+        public AugmentQualityFromReleaseName(IHistoryService historyService)
+        {
+            _historyService = historyService;
+        }
+
+        public AugmentQualityResult AugmentQuality(LocalEpisode localEpisode, DownloadClientItem downloadClientItem)
+        {
+            // Don't try to augment if we can't lookup the grabbed history by downloadId
+            if (downloadClientItem == null)
+            {
+                return null;
+            }
+
+            var fileQuality = localEpisode.FileEpisodeInfo?.Quality.Quality;
+            var folderQuality = localEpisode.FolderEpisodeInfo?.Quality.Quality;
+            var localQuality = folderQuality ?? fileQuality;
+
+            // Return early if the file or folder quality is not a television source (preferring the folder over the file)
+            if (localQuality?.Source != QualitySource.Television)
```

Comment:
> If the filename contains HDTV then it must be HDTV. If the filename contains WEB-DL then it's WEB-DL. Both is Confidence=Tag and handled by those augmenters.
They should win, even if the indexer says otherwise. The indexer should only be used if the source isn't specified in the filename.

The problem is we don't have a differentiation there. `QualityParser` will return `HDTV720p` even if it only has 720p in the filename.

> Previously this would've been HDTV, and blocked on import. Now it'll be imported as bluray. I feel that might be a bit too eager.

Previously (before last week) it'd import as whatever quality the filename said it was, then if the same release with a different name was grabbed as an upgrade it'd reject that as having the same file size as existing.

> Example: [CR] Sailor Moon - 004 [480p][48CE2D0F] testcase. Resolution clearly is 480p. QualityDetectionSource will be 'Name' because of that, even though it's entirely unknown what the Source really is. We can guess, but we'd trust the release title over that guess.

For a lot of anime the source of this would actually be WEB too unfortunately.

> I don't think any of this is truly fixable without splitting up Source & Resolution in QualityModel into Source, Resolution, SourceConfidence, ResolutionConfidence and EffectiveQuality. This would allow QualityParser to specify what it parsed explicitly and what it inferred.
Leaving it to the aggregators to decide what information to trust the most.

Yeah, that's the crux of it.

> PS: A temporary workaround could be to have the aggregators do a secondary parse of the actual name, if information between indexer & filename/mediainfo conflicts. Saving us from having to change QualityModel throughout the program and the existing items in the history table.

This seems like the easiest way to clear this up and not block future improvements.